frontend/src/Components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,215 @@
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Manager, Popper, Reference } from 'react-popper';
+import Portal from 'Components/Portal';
+import { kinds, tooltipPositions } from 'Helpers/Props';
+import dimensions from 'Styles/Variables/dimensions';
+import { isMobile as isMobileUtil } from 'Utilities/browser';
+import styles from './Tooltip.css';
+
+export interface TooltipProps {
+  className?: string;
+  bodyClassName?: string;
+  anchor: React.ReactNode;
+  tooltip: string | React.ReactNode;
+  kind?: Extract<(typeof kinds.all)[number], keyof typeof styles>;
+  position?: (typeof tooltipPositions.all)[number];
+  canFlip?: boolean;
+}
+function Tooltip(props: TooltipProps) {
+  const {
+    className,
+    bodyClassName = styles.body,
+    anchor,
+    tooltip,
+    kind = kinds.DEFAULT,
+    position = tooltipPositions.TOP,
+    canFlip = false,
+  } = props;
+
+  const closeTimeout = useRef(0);
+  const updater = useRef<(() => void) | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleClick = useCallback(() => {
+    if (!isMobileUtil()) {
+      return;
+    }
+
+    setIsOpen((isOpen) => {
+      return !isOpen;
+    });
+  }, [setIsOpen]);
+
+  const handleMouseEnter = useCallback(() => {
+    // Mobile will fire mouse enter and click events rapidly,
+    // this causes the tooltip not to open on the first press.
+    // Ignore the mouse enter event on mobile.
+    if (isMobileUtil()) {
+      return;
+    }
+
+    if (closeTimeout.current) {
+      window.clearTimeout(closeTimeout.current);
+    }
+
+    setIsOpen(true);
+  }, [setIsOpen]);
+
+  const handleMouseLeave = useCallback(() => {
+    // Still listen for mouse leave on mobile to allow clicks outside to close the tooltip.
+
+    setTimeout(() => {
+      setIsOpen(false);
+    }, 100);
+  }, [setIsOpen]);
+
+  const maxWidth = useMemo(() => {
+    const windowWidth = window.innerWidth;
+
+    if (windowWidth >= parseInt(dimensions.breakpointLarge)) {
+      return 800;
+    } else if (windowWidth >= parseInt(dimensions.breakpointMedium)) {
+      return 650;
+    } else if (windowWidth >= parseInt(dimensions.breakpointSmall)) {
+      return 500;
+    }
+
+    return 450;
+  }, []);
+
+  const computeMaxSize = useCallback(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (data: any) => {
```

Comment:
Only type I could get working is this this `react-popper` doesn't export much from `popper`

Not great 🙁 

```suggestion
  const computeMaxSize = useCallback<
    NonNullable<
      NonNullable<
        NonNullable<PopperProps['modifiers']>['computeMaxHeight']
      >['fn']
    >
  >(data) => {
```

Reply:
Yeah, saw there wasn't much exported and gave up. I ran into issues with this change and reverted back, but not too worried, eventually popper will get updated/replaced and it'll improve.